ADR/user-specified-tokens-build.md

```diff
@@ -0,0 +1,139 @@
+# Enable users to configure tokens/credentials for a Build/Test Pipeline
+
+Date: 2023-03-03
+
+## Status
+
+Provisional
+
+## Context
+
+As a user, I would like to configure sensitive information/tokens/credentials for the build/test pipeline to consume as part of a TaskRun.
+
+This enables partner tasks contributed by Red Hat partners ([STONE-549](https://issues.redhat.com/browse/STONE-549)), but it also enables some tasks in our existing build pipeline like the [sast-snyk task in our build-definitions repo](https://github.com/redhat-appstudio/build-definitions/blob/main/task/sast-snyk-check/0.1/sast-snyk-check.yaml) ([STONE-415](https://issues.redhat.com/browse/STONE-415)).
+
+## Decision
+
+### How would the sensitive information be persisted and made available to the PipelineRun ?
+
+1. The UI would take in sensitive information as form input and [upload it to SPI](https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#uploading-access-token-to-spi-using-kubernetes-secret) using the user's authentication token.
+
+
+| Name of the Secret   | Token data |
+| ----------- | ----------- |
+| synk-secret      | *user-provided*   |
+| tidelift-token   | *user-provided*   |
+| *user-provided*  | *user-provided*   |
+
+
+2. Subsequently, the UI would create an `SPIAccessTokenBinding` CR with the appropriate name in the `.spec.secret.name` such that the secret appears under a well-known name. The well-known name would be specified in the Task definition by the Task author.
+
+3. The PipelineRun must reference the secret by name.
+
+
+### Do the build-service and integration service controllers directly reference the Secret?
+
+They do not.
+
+### Example
+
+The contents of the `Secret` would be 'uploaded' using a temporary `Secret`. 
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-upload-secret
+  labels:
+    spi.appstudio.redhat.com/upload-secret: token
+    spi.appstudio.redhat.com/token-name: my-spi-access-token-binding-for-snyk
+type: Opaque
+stringData:
+  spiTokenName: my-spi-access-token-binding-for-snyk
+  providerUrl: https://snyk.io/
+  # The username is irrelevant, but has to be here for SPI to process this
+  userName: my-username-goes-here
```

Comment:
So UI should add a random username here?

Reply:
Yes.